include/zephyr/pm/state.h

```diff
@@ -339,6 +361,27 @@ struct pm_state_info {
  */
 uint8_t pm_state_cpu_get_all(uint8_t cpu, const struct pm_state_info **states);
 
+/**
+ * Retrieve the index of a CPU's DT cpu-power-states array element.
+ *
+ * @param cpu CPU index.
+ * @param state pm_state state.
+ * @param substate_id pm_state substate ID.
+ *
+ * @return Index of the PM state, or -1.
+ */
+uint8_t pm_state_cpu_get_index(uint8_t cpu, enum pm_state state, uint8_t substate_id);
```

Comment:
this function is not needed, the only one used is `pm_state_get_index`

Reply:
Nice catch. I think we should probably drop `pm_state_get_index` and use the CPU-specific one instead, just to match [the call here](https://github.com/zephyrproject-rtos/zephyr/pull/60998/files#diff-d4e51fe3d6c0c3c82cef6a93687ea6246bc506c89f069f071138be1d497459d6R143-R144) when we pass the CPU-specific sub/state for which we want to retrieve the index.
On single-core platforms the `#if`-guarded variant would just drop the CPU id. Otherwise, its current state this could lead to discrepancies and notifiers not triggering if cores have diverging `cpu-power-states` definitions.